src/plugins/part.c

```diff
@@ -666,6 +666,100 @@ gboolean bd_part_set_part_flag (const gchar *disk, const gchar *part, BDPartFlag
 }
 
 /**
+ * bd_part_set_part_flags:
+ * @disk: disk the partition belongs to
+ * @part: partition to set the flag on
+ * @flags: flags to set (mask combined from #BDPartFlag numbers)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the @flags were successfully set on the @part partition or
+ *          not
+ *
+ * Note: Unsets all the other flags on the partition.
+ *
+ */
+gboolean bd_part_set_part_flags (const gchar *disk, const gchar *part, guint64 flags, GError **error) {
+    PedDevice *dev = NULL;
+    PedDisk *ped_disk = NULL;
+    PedPartition *ped_part = NULL;
+    const gchar *part_num_str = NULL;
+    gint part_num = 0;
+    guint64 i = 0;
+    gint status = 0;
+    gboolean ret = FALSE;
+
+    /* TODO: share this code with the other functions modifying a partition */
+    if (!part || (part && (*part == '\0'))) {
+        g_set_error (error, BD_PART_ERROR, BD_PART_ERROR_INVAL,
+                     "Invalid partition path given: '%s'", part);
+        return FALSE;
+    }
+
+    dev = ped_device_get (disk);
+    if (!dev) {
+        set_parted_error (error, BD_PART_ERROR_INVAL);
+        g_prefix_error (error, "Device '%s' invalid or not existing", disk);
+        return FALSE;
+    }
+
+    ped_disk = ped_disk_new (dev);
+    if (!ped_disk) {
+        set_parted_error (error, BD_PART_ERROR_FAIL);
+        g_prefix_error (error, "Failed to read partition table on device '%s'", disk);
+        ped_disk_destroy (ped_disk);
```

Comment:
Is it necessary to call `ped_disk_destroy (ped_disk);` here since ped_disk is NULL?